src/HistoryQuery/HistoryQuery.class.js

```diff
@@ -271,7 +291,13 @@ class HistoryQuery {
       }
 
       // eslint-disable-next-line no-await-in-loop
-      await this.south.historyQuery(scanMode, startTime, intervalEndTime)
+      const historyQueryResult = await this.south.historyQuery(scanMode, startTime, intervalEndTime)
+
+      if (historyQueryResult === -1) {
+        this.logger.error(`Error while retrieving data. Exiting historyQueryHandler. startTime: ${
+          startTime.toISOString()}, intervalEndTime: ${intervalEndTime.toISOString()}`)
+        return -1
```

Comment:
return -1 or 0 ? should we return true/false instead?

Reply:
I initially planned to add other error codes, but ended with only two. I prefer specifying -1 since some method returns the number of values so there won't be any chance to mistake a return value true/false with a number (1/0)